File: BaekJoon/DFS, BFS/2178_미로탐색.py
def bfs(map_list, result, n, m):
    dx = [1, -1, 0, 0]
    dy = [0, 0, 1, -1]
    
    result[0][0] = 1
    
    queue= []
    queue.append([0,0])
    
    while queue:
        x, y = queue.pop(0)
        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]
            if nx >= 0 and nx < n and ny >= 0 and ny < m:
                if map_list[nx][ny] == 1 and result[nx][ny] == 0:
                    queue.append([nx,ny])
                    result[nx][ny] = result[x][y] + 1
    
    return result[n-1][m-1]

def main():
    n, m = map(int, input().split())
    
    map_list = [list(map(int, input())) for _ in range(n)]
    result = [[0 for _ in range(m)] for _ in range(n)]
 
    print(bfs(map_list, result, n, m))
    
if __name__ == "__main__":
    main()

File: BaekJoon/DFS, BFS/2178_미로탐색.swift
//
//  main.swift
//  AlgorithmCommandLine
//
//  Created by 고도 on 2022/12/16.
//

import Foundation

let input = readLine()!.split(separator: " ").map { Int(String($0))! }
var maps: [[Int]] = []
var check: [[Int]] = [[Int]](repeating: [Int](repeating: 0, count: input[1]), count: input[0])

for _ in 0..<input[0] {
    let info = readLine()!.map { Int(String($0))! }
    maps.append(info)
}

move()

print(check[input[0] - 1][input[1] - 1])

func move() {
    var queue: [(Int, Int)] = []
    
    queue.append((0, 0))
    check[0][0] = 1
    
    while !queue.isEmpty {
        let e = queue.removeFirst()
        
        let dx = [0, 0, -1, 1]
        let dy = [-1, 1, 0, 0]
        
        for i in 0..<4 {
            let nx = e.0 + dx[i]
            let ny = e.1 + dy[i]
            
            if 0 <= nx, nx < input[0], 0 <= ny, ny < input[1] {
                if maps[nx][ny] == 1, check[nx][ny] == 0 {
                    check[nx][ny] = check[e.0][e.1] + 1
                    queue.append((nx, ny))
                }
            }
        }
    }
}